src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -947,9 +947,14 @@ void CmdSketcherCompCreateBSpline::activated(int iMsg)
 {
     if (iMsg == 0) {
         ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerBSpline(iMsg));
-    } else if (iMsg == 1) {
+    }
+    else if (iMsg == 1) {
         ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerBSpline(iMsg));
-    } else {
+    }
+    else if (iMsg == 2) {
+        ActivateHandler(getActiveGuiDocument(), new DrawSketchHandlerBSpline(iMsg));
+    }
+    else {

```

Comment:
@abdullahtahiriyo I could use some input here. I can either extend `DrawSketchHandlerSpline` as I'm doing now, or create a new DSH especially for interpolation. Both have similarilities in how they need to be input, but there's a major difference that when defining poles we draw circles, and otherwise we draw vertices.

Reply:
Both are possible. If the differences are significant, I think you would be better served by a new DSH. Each individual DSH will be simpler and easier to read and understand. Future developers reading the code will also be happier.

Reply:
My only concern is that making a different class could result in code duplication. I guess we can work on that later, reducing the duplication rather than trying to split documents.

Reply:
It is always a trade-off. A DSH does not have that many parts. If the mouse-flow and representation is different and the command execution is different there is about nothing in common. But, yes, we can look to code duplication and what exactly to do about it when we know how much we are actually duplicating, if any.

As an example of what happens when a lot of adaptations of behaviour and command execution are implemented in a DSH, you may look at the lineset DSH (polyline). It is implemented as an all-in-one DSH. The complexity frightens developers eyeing it.

Reply:
The current DSH framework offers support for different creation modes, which is the parameter provided on activation. If the input were the same, (for example an array of points, each point taken after one click) one could use a generic name for that array (e.g. inputPoints), without indicating if it is poles or interpolation points. Then it is possible to reuse all the input code. The command execution will be different. Two different member functions, called depending on the creation mode can be created and called appropriately. Small exceptions may be made for some differentiation between modes (e.g. less amount of steps,). But I think that is about the limit.

So, yes, we want to do code reuse in every instance where it makes sense. We want to keep separate things that change for different reasons (not unexpectedly introducing bugs on unrelated functionality while fixing a bug, just because it is tightly coupled despite being a separate thing). At the end, we want the code to be maintainable which is kind of the setting of the trade-off. We want to avoid duplication because duplicating creates problems to maintainability, we want to keep unrelated functionality separate, because mixing unrelated functionality creates problems to maintainability. There is where the skill of the developer enters into play ;)